integrations/dbt_bigquery/dbt/adapters/openlineage_bigquery/connections.py

```diff
@@ -0,0 +1,284 @@
+import re
+import uuid
+from contextlib import contextmanager
+from datetime import datetime
+from typing import Optional, Tuple
+
+import agate
+import google.cloud.exceptions
+import google.auth.exceptions
+from dataclasses import dataclass
+
+from dbt.adapters.bigquery.connections import BigQueryAdapterResponse, BigQueryCredentials, \
+    BigQueryConnectionManager
+from dbt.adapters.openlineage_bigquery.version import __version__ as VERSION
+from dbt.clients import agate_helper
+from dbt.exceptions import DatabaseException, RuntimeException
+from dbt.logger import GLOBAL_LOGGER as logger
+from dbt.utils import format_bytes, format_rows_number
+
+from marquez.provider.bigquery import BigQueryDatasetsProvider
+from marquez.sql import SqlParser
+from openlineage.client import OpenLineageClientOptions, OpenLineageClient
+from openlineage.facet import SourceCodeLocationJobFacet, SqlJobFacet
+from openlineage.run import RunEvent, RunState, Run, Job, Dataset
+
+
+@dataclass
+class OpenLineageBigQueryCredentials(BigQueryCredentials):
+    """setting openlineage_url in dbt profile is required"""
+    openlineage_url: Optional[str] = None
+    openlineage_timeout: float = 5.0
+    openlineage_apikey: Optional[str] = None
+
+    @property
+    def type(self) -> str:
+        return 'openlineage_bigquery'
+
+
+@dataclass
+class RunMeta:
+    """Runtime information about model set up on start event that
+    needs to be shared with complete/fail events
+    """
+    run_id: str
+    namespace: str
+    name: str
```

Comment:
Is this the job name?

Reply:
Yes. It references output table, for example: `model.dbt_openlineage_test.test_third_dbt_model`

Reply:
I think it would be helpful prefix job related fields with `job_`. It's not obvious that name refers to the job name. (Not a locker to merge though)